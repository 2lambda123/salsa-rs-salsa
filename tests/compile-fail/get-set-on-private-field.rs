#[salsa::jar(db = Db)]
pub struct Jar(a::MyInput);

mod a {
    #[salsa::input(jar = crate::Jar)]
    pub struct MyInput {
        field: u32,
    }
}

pub trait Db: salsa::DbWithJar<Jar> {}

#[salsa::db(Jar)]
#[derive(Default)]
struct Database {
    storage: salsa::Storage<Self>,
}

impl salsa::Database for Database {}

impl Db for Database {}

fn main() {
    let mut db = Database::default();
    let input = a::MyInput::new(&mut db, 22);

    input.field(&db);
    input.set_field(&mut db).to(23);
}
