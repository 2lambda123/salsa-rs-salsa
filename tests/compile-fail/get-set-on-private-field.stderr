error[E0405]: cannot find trait `__salsa_crate_Db` in module `self`
 --> tests/compile-fail/get-set-on-private-field.rs:8:16
  |
8 |     pub struct MyInput {
  |                ^^^^^^^ not found in `self`
  |
help: consider importing this trait through its public re-export
  |
5 +     use crate::__salsa_crate_Db;
  |
